peps/pep-0783.rst

```diff
@@ -0,0 +1,189 @@
+ABI-sensitive flags.
+
+
+Specification
+=============
+
+The platform tags will take the form::
+
+    pyodide_${YEAR}_${PATCH}_wasm32
+
+Each one of these will be used with a specified Python version. For example, the
+platform tag ``pyodide_2025_0`` will be used with Python 3.13.
+
+Emscripten Wheel ABI
+--------------------
+
+The specification of the ``pyodide_<abi>`` platform includes:
+
+* Which version of the Emscripten compiler is used
+* What libraries are statically linked with the interpreter
+* What stack unwinding ABI is to be used
+* Which runtime platform features are required to be present
+
+and a handful of other similar details that affect the ABI.
+
+The ABI is selected by choosing the appropriate version of the Emscripten
+compiler and passing appropriate compiler and linker flags. It is possible for
+other people to build their own Python interpreter that is compatible with the
+Pyodide ABI, it is not necessary to use the Pyodide distribution itself.
+
+The ``pyodide build`` tool knows how to create wheels that match our ABI. Unlike
+with manylinux wheels, there is no need for a Docker container to build the
+``pyodide_<abi>`` wheels. All that is needed is a Linux machine and appropriate
+versions of Python, Node.js, and Emscripten.
+
+It is possible to validate a wheel by installing and importing it into the
+Pyodide runtime. Because Pyodide can run in an environment with strong
+sandboxing guarantees, doing this produces no security risks.
+
+Determining the ABI version
+---------------------------
+
+The Pyodide ABI version is stored in the ``PYODIDE_ABI_VERSION`` config variable
+and can be determined via::
+
+    pyodide_abi_version = sysconfig.get_config_var("PYODIDE_ABI_VERSION")
+
+
+To generate the list of compatible tags, one can use the following code::
+
+    from packaging.tags import cpython_tags, _generic_platforms
+
+    def _emscripten_platforms() -> Iterator[str]:
+        pyodide_abi_version = sysconfig.get_config_var("PYODIDE_ABI_VERSION")
+        if pyodide_abi_version:
+            yield f"pyodide_{pyodide_abi_version}_wasm32"
+        yield from _generic_platforms()
+
+    emscripten_tags = cpython_tags(platforms=_emscripten_platforms())
+
+This code will be added to `pypa/packaging
+<https://github.com/pypa/packaging/pull/804>`__.
+
+
+Package Installers
+------------------
+
+Installers should use the ``_emscripten_platforms()`` function shown above to
+determine which platforms are compatible with an Emscripten build of CPython. In
+particular, the Pyodide ABI version is exposed via
+``sysconfig.get_config_var("PYODIDE_ABI_VERSION")``.
+
+Package indexes
+---------------
+
+We recommend that package indexes accept any wheel whose platform tag matches
+``pyodide_[0-9]+_[0-9]+_wasm32``.
+
+
+Dependency Specifier Markers
+----------------------------
+
+To check for the Emscripten platform in a dependency specifier, one can use
+``sys_platform == 'emscripten'`` (or its negation).
```

Comment:
Perhaps link to the formal specification here?

Reply:
Done in 73d0bfc.